proxies/dotnet/Controllers/Location.cs

```diff
@@ -0,0 +1,189 @@
+                } else if (param["value"].Type == JTokenType.String) {
+                    result.Add(param["value"].Value<string>() ?? "");
+                } else if (param["value"].Type == JTokenType.Object) {
+                    result.Add(param["value"]);
+                } else {
+                    result.Add(param["value"].Value<bool>());
+                }
+            }
+        }
+        return result;
+    }
+
+    private object GetEntity(string location) {
+        var parts = location.Split('/');
+        var type = parts[0];
+        var id = parts[1];
+
+        object? result = null;
+        if (type == "client") {
+           if (DataStore.CloudClients.TryGetValue(id, out DVCCloudClient? cloudClient)) {
+                result = cloudClient;
+           } else if (DataStore.LocalClients.TryGetValue(id, out DVCLocalClient? localClient)) {
+                result = localClient;
+           }
+        } else if (type == "user" && DataStore.Users.TryGetValue(id, out User? user)) {
+            result = user;
+        } else if (type == "command" && DataStore.Commands.TryGetValue(id, out object? command)) {
+            result = command;
+        }
+        if (result == null) {
+            throw new Exception($"Entity {location} not found");
+        }
+        return result;
+    }
+
+    private async Task<CommandResult> InvokeCommand(object entity, string command, bool isAsync, List<object> parsedParams) {
+        var parsedCommand = char.ToUpper(command[0]) + command.Substring(1);
+        parsedCommand = entity is DVCCloudClient ? parsedCommand + "Async" : parsedCommand;
+
+        MethodInfo? commandMethod = entity.GetType().GetMethod(parsedCommand);
+        if (command == "variable") {
+            Type defaultValueClass = parsedParams[parsedParams.Count - 1].GetType();
+            commandMethod = commandMethod?.MakeGenericMethod(defaultValueClass); // have to set the generic type for defaultValue before invoke
+        }
+
+        object result = new {};
+
+        if (isAsync) {
+            dynamic? task = commandMethod?.Invoke(entity, parsedParams.ToArray());
+            result = (await task) ?? result;
+        } else {
+            result = commandMethod?.Invoke(entity, parsedParams.ToArray()) ?? result;
+        }
+
+        var resultId = DataStore.Commands.Count.ToString();
+        if (result != null) DataStore.Commands.Add(resultId, result);
+
+        var type = result != null ? GetEntityType(result) : "Void";
+
+        return new CommandResult {
+            EntityType = type,
+            Data = result ?? new {},
+            CommandId = resultId
+        };
+    }
+
+    private string GetEntityType(object result) {
+        var type = TypeDescriptor.GetClassName(result) ?? "Object";
+        if (type.Contains("Dictionary")) {
+            type = "Object";
+        } else {
+            type = type.Split(".").Last();
+        }
+
+        return EntityTypes.Contains(type) ? type : "Void";
```

Comment:
I think "Void" is supposed to represent "nothing was returned" not "we dont know what type this corresponds to". That case should be "Object"

Reply:
yeah, really not sure why this is being done. I think it shouldnt be object either tbh, object should already be covered. This method shoulld probably just return type

Reply:
im just gonna make it object for now to appease the enum gods